### ISSUE TITLE
Prefering connecting to onion server "if" EC is using Tor as proxy

### ISSUE THREAD

**Issue by HostFat** [Electron-Cash/Electron-Cash#1179]
As #1178 has been added, maybe it can be useful to set a "preference" to connect to electrumx servers on onion addresses.

If the onion server(s) doesn't work/answer, EC will then try with the other common servers.

**Comment by cculianu** [comment]
Yes, all's well and good --  but all this logic add to the giant flying spaghetti monster that is the network layer.  :)

I will definitely do something along these lines.  The exact approach compatible with this pasta al'arrabiata remains to be seen. :)
